demosplan/DemosPlanCoreBundle/Logic/Statement/StatementService.php

```diff
@@ -1522,35 +1525,40 @@ protected function updateStatementArray(array $data)
     /**
      * Add a report entry to the DB.
      *
-     * @param string $procedureId
-     * @param array  $accessMap
-     * @param string $statementId
+     * @var StatementViewed[]
      *
      * @throws ORMException
      * @throws OptimisticLockException
      * @throws ReflectionException
      */
-    private function addStatementViewedReport($procedureId, $accessMap, $statementId): void
+    private function addStatementViewedReport(array $viewsToLog): void
     {
-        // only log if user is known
-        if (!\array_key_exists('user', $accessMap)) {
-            return;
-        }
-        $alreadyLogged = $this->reportService
-            ->statementViewLogged($procedureId, $accessMap['user'], $statementId);
+        $entries = [];
+        foreach ($viewsToLog as $viewedEntry) {
+            $procedureId = $viewedEntry->getProcedureId();
+            $accessMap = $viewedEntry->getAccessMap();
+            $statementId = $viewedEntry->getStatementId();
 
-        // logging access once is enough
-        if ($alreadyLogged) {
-            return;
-        }
+            // only log if user is known
+            if (!\array_key_exists('user', $accessMap)) {
+                return;
+            }
+            $alreadyLogged = $this->reportService
+                ->statementViewLogged($procedureId, $accessMap['user'], $statementId);
 
-        $entry = $this->statementReportEntryFactory->createViewedEntry(
-            $statementId,
-            $procedureId,
-            $accessMap
-        );
+            // logging access once is enough
+            if ($alreadyLogged) {
+                return;
+            }
 
-        $this->reportService->persistAndFlushReportEntries($entry);
+            $entries[] = $this->statementReportEntryFactory->createViewedEntry(
+                $statementId,
+                $procedureId,
+                $accessMap
+            );
+        }
+
+        $this->reportService->persistAndFlushReportEntries($entries);
```

Comment:
can u please check this one more time, persistAndFlushReportEntries expects a parameter of type ReportEntry. It is not working with an array $entries

Reply:
Did you try it? The method extracts the array and then expects ReportEntries. It worked for me

Reply:
yup i tested it and retested it and still not working with an error that indicate that  a wrong type was given to persistAndFlushReportEntries. 
![image](https://github.com/user-attachments/assets/159bc119-aff6-4f95-ba37-02571d179bff)

Reply:
This is only the typehint of the IDE. What is the error?

Reply:
here is the error message, i remove all cache and still get it. When i add array as typehint for reportEntries it works
![image](https://github.com/user-attachments/assets/29582dca-6e47-437b-813b-3c03eedf77ed)

Reply:
which php version do you use?

Reply:
here is the error message, i remove all cache and still get it. When i add array as typehint for reportEntries it works 
![image](https://github.com/user-attachments/assets/b17b43d5-df5a-4ef4-92cc-7d65f3680570)

Reply:
8.3

Reply:
really strange, it still works for me. I pushed a fix, could you please try whether this fixes the issue for you @bakridemos ?